compiler/src/yul/selectors.rs

```diff
@@ -0,0 +1,175 @@
+use crate::errors::CompileError;
+use crate::yul::base;
+use tiny_keccak::{Hasher, Keccak};
+use yultsur::yul;
+
+/// Builds a switch statement from the contract ABI.
+/// The switch's expression is the 4 left-most bytes in the calldata and each case is
+/// defined as the keccak value of each function's signature (without return data).
+pub fn switch(functions: Vec<&ethabi::Function>) -> Result<yul::Switch, CompileError> {
+    let cases = functions
+        .into_iter()
+        .map(|function| case(function))
+        .collect::<Result<Vec<yul::Case>, CompileError>>()?;
+
+    Ok(yul::Switch {
+        expression: expression(),
+        cases,
+    })
+}
+
+/// Builds an expression that loads the first 4 bytes of the calldata.
+pub fn expression() -> yul::Expression {
```

Comment:
do we need anything `pub` in this module except `switch`?

Reply:
Nope :)

https://github.com/g-r-a-n-t/rust-vyper/blob/types/compiler/src/yul/runtime/abi.rs